examples/niscope/plot-read-waveform.py

```diff
@@ -58,68 +68,79 @@ def CheckStatus(scope_service, vi, result):
-    num_records = 1,
-    enforce_realtime = True
-))
-CheckStatus(scope_service, vi, config_result)
-
-# Configure vertical timing
-vertical_result = scope_service.ConfigureVertical(niscope_types.ConfigureVerticalRequest(
-    vi = vi,
-    channel_list = channels,
-    range = 10.0,
-    offset = 0,
-    coupling = niscope_types.VerticalCoupling.VERTICAL_COUPLING_NISCOPE_VAL_DC,
-    enabled = True,
-    probe_attenuation = 1
-))
-CheckStatus(scope_service, vi, vertical_result)
-
-confTrigger_edge_result = scope_service.ConfigureTriggerEdge(niscope_types.ConfigureTriggerEdgeRequest(
-    vi = vi,
-    trigger_source = channels,
-    level = 0.00,
-    trigger_coupling = niscope_types.TriggerCoupling.TRIGGER_COUPLING_NISCOPE_VAL_DC,
-    slope = niscope_types.TriggerSlope.TRIGGER_SLOPE_NISCOPE_VAL_POSITIVE
-))
-CheckStatus(scope_service, vi, confTrigger_edge_result)
-
-result = scope_service.SetAttributeViInt32(niscope_types.SetAttributeViInt32Request(
-    vi = vi,
-    channel_list = channels,
-    attribute_id = niscope_types.NiScopeAttributes.NISCOPE_ATTRIBUTE_MEAS_REF_LEVEL_UNITS,
-    value = niscope_types.NiScopeInt32AttributeValues.NISCOPE_INT32_REF_LEVEL_UNITS_VAL_VOLTS
-))
-CheckStatus(scope_service, vi, result)
-
-# Read a waveform from the scope
-read_result = scope_service.Read(niscope_types.ReadRequest(
-    vi = vi,
-    channel_list = channels,
-    timeout = 10000,
-    num_samples = 100000
-))
-CheckStatus(scope_service, vi, read_result)
-values = read_result.waveform[0:10]
-print(values)
-
-scope_service.Close(niscope_types.CloseRequest(
-    vi = vi
-))
-channel.close()
+    CheckForError(vi, confTrigger_edge_result.status)
+
+    result = scope_service.SetAttributeViInt32(niscope_types.SetAttributeViInt32Request(
+        vi = vi,
+        channel_list = channels,
+        attribute_id = niscope_types.NiScopeAttributes.NISCOPE_ATTRIBUTE_MEAS_REF_LEVEL_UNITS,
+        value = niscope_types.NiScopeInt32AttributeValues.NISCOPE_INT32_REF_LEVEL_UNITS_VAL_VOLTS
+    ))
+    CheckForError(vi, result.status)
+
+    # Read a waveform from the scope
+    read_result = scope_service.Read(niscope_types.ReadRequest(
+        vi = vi,
+        channel_list = channels,
+        timeout = 10000,
+        num_samples = 100000
+    ))
+    CheckForError(vi, read_result.status)
+    values = read_result.waveform[0:10]
+    print(values)
+
+except grpc.RpcError as rpc_error:
+    error_message = rpc_error.details()
+    if rpc_error.code() == grpc.StatusCode.UNAVAILABLE :
+        error_message = f"Failed to connect to server on {server_address}"
+    elif rpc_error.code() == grpc.StatusCode.UNIMPLEMENTED:
+        error_message = "The operation is not implemented or is not supported/enabled in this service"
+    print(f"{error_message}")
+
+finally:
+    if('vi' in vars() and vi.id != 0):
+        # close the session.
+        CheckForError(vi, (scope_service.Close(niscope_types.CloseRequest(
+            vi = vi
+        ))).status)
```

Comment:
In this file also, try-except block, more exception handling and finally-block is added. Mentioning here because the diff could be misleading.